sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -134,45 +134,50 @@ class TreeGrower:
 
     Parameters
     ----------
-    X_binned : ndarray of int, shape (n_samples, n_features)
+    X_binned : ndarray of int of shape (n_samples, n_features)
         The binned input samples. Must be Fortran-aligned.
-    gradients : ndarray, shape (n_samples,)
+    gradients : ndarray of shape (n_samples,)
         The gradients of each training sample. Those are the gradients of the
         loss w.r.t the predictions, evaluated at iteration ``i - 1``.
-    hessians : ndarray, shape (n_samples,)
+    hessians : ndarray of shape (n_samples,)
         The hessians of each training sample. Those are the hessians of the
         loss w.r.t the predictions, evaluated at iteration ``i - 1``.
-    max_leaf_nodes : int or None, optional (default=None)
+    max_leaf_nodes : int, default=None
         The maximum number of leaves for each tree. If None, there is no
         maximum limit.
-    max_depth : int or None, optional (default=None)
+    max_depth : int, default=None
         The maximum depth of each tree. The depth of a tree is the number of
         edges to go from the root to the deepest leaf.
         Depth isn't constrained by default.
-    min_samples_leaf : int, optional (default=20)
+    min_samples_leaf : int, default=20
         The minimum number of samples per leaf.
-    min_gain_to_split : float, optional (default=0.)
+    min_gain_to_split : float, default=0.
         The minimum gain needed to split a node. Splits with lower gain will
         be ignored.
-    n_bins : int, optional (default=256)
+    n_bins : int, default=256
         The total number of bins, including the bin for missing values. Used
         to define the shape of the histograms.
-    n_bins_non_missing_ : array of uint32
+    n_bins_non_missing : ndarray, dtype=np.uint32, default=None
         For each feature, gives the number of bins actually used for
         non-missing values. For features with a lot of unique values, this
         is equal to ``n_bins - 1``. If it's an int, all features are
         considered to have the same number of bins. If None, all features
         are considered to have ``n_bins - 1`` bins.
-    has_missing_values : ndarray of bool or bool, optional (default=False)
+    has_missing_values : bool or ndarray of bool, default=False
```

Comment:
Although the data type of `ndarray` is specified with `dtype`, I do not know if:

```suggestion
    has_missing_values : bool or ndarray, dtype=bool, default=False
```

is better than the current approach.